deps.ffmpeg/patches/libdatachannel/0001-one-symlink-only-for-library.patch

```diff
@@ -0,0 +1,21 @@
+XCode only resolves one level of symlinks [0].
+
+By default libdatachannel generates three layers. This patch
+reduces that to two.
+
+[0] https://stackoverflow.com/questions/29946961/xcode-copy-files-build-phase-and-symlinks
+
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 2ba612d8..1f4348d1 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -262,7 +262,7 @@ add_library(datachannel SHARED
+ 	${LIBDATACHANNEL_IMPL_HEADERS})
+ set_target_properties(datachannel PROPERTIES
+ 	VERSION ${PROJECT_VERSION}
+-	SOVERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}
++	SOVERSION ${PROJECT_VERSION}
+ 	CXX_STANDARD 17
+ 	CXX_VISIBILITY_PRESET default)
+ target_compile_definitions(datachannel PRIVATE RTC_EXPORTS)
```

Comment:
Why are we patching libdatachannel? Can this be upstreamed?

Reply:
@paullouisageneau

What do you think? The change adding minor version to the library breaks Xcode when copying during build. What is your take on reverting?

It seems unfortunate to make libdatachannel worse because of Xcode.

Reply:
https://github.com/paullouisageneau/libdatachannel/commit/7591b9695ccf6d6aea1b3e84e0a17de3c448bca3 Is the commit that would be reverted.

Reply:
The thing is that the library is ABI-compatible between build numbers, to `${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}` is indeed what you want here, otherwise dynamic library compatibility would be broken.

Reply:
cc @PatTheMav who understands macOS and Xcode requirements far better than I do.

Reply:
> The thing is that the library is ABI-compatible between build numbers, to `${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}` is indeed what you want here, otherwise dynamic library compatibility would be broken.

Library ABI compatibility is not handled via file names but headers inside the library on macOS - the link loader will check the version and compatibility version of the loaded library against the requirements recorded inside the binary and will abort loading the library if the versions don’t match.

Using versioned file names does not fulfil that requirement and is not supported by macOS (e.g. it will not search for or load a less specific variant of the versioned file name).

Reply:
The correct way to build a shared library for macOS requires:

* Setting the `VERSION` to `0`
* Setting the `SOVERSION` to `0`
* Setting the `MACHO_COMPATIBILITY_VERSION` (increment this when breaking ABI)
* Setting the `MACHO_CURRENT_VERSION` to the actual version
* Setting `NO_SONAME` to `TRUE` to disable versioned filenames

With those target properties set, you'll generate macOS-style dynamic libraries that work with the native dynamic library loader system and the native build tools.

Reply:
@PatTheMav Should I send a PR to libdatachannel putting that behind a platform check? Would you mind reviewing it?

Reply:
> @PatTheMav Should I send a PR to libdatachannel putting that behind a platform check? Would you mind reviewing it?

Sure.

Reply:
Created PR here https://github.com/paullouisageneau/libdatachannel/pull/1091

Reply:
To close this thread out, I would like to see this patch updated to what was ultimately merged into libdatachannel, or have us update to a new tag that contains the change (presumably either v0.19.x or v0.20.0).

Reply:
@RytoEX I am going to update to upstream (no patch required!)

@paullouisageneau is that a good idea? Or is a v.20 a long way away?

Reply:
FYI I just released libdatachannel v0.20.0

Reply:
Patches removed as a result of bumping the version to v0.20.1, so this is moot. Updating to v0.20.1 has its own concerns to address separately.